vortex-array/src/array/chunked/mod.rs

```diff
@@ -38,38 +40,48 @@ impl ChunkedArray {
             }
         }
 
-        let chunk_ends = PrimitiveArray::from_vec(
-            [0u64]
-                .into_iter()
-                .chain(chunks.iter().map(|c| c.len() as u64))
-                .scan(0, |acc, c| {
-                    *acc += c;
-                    Some(*acc)
-                })
-                .collect_vec(),
-            NonNullable,
-        );
+        let chunk_ends = [0u64]
+            .into_iter()
+            .chain(chunks.iter().map(|c| c.len() as u64))
+            .scan(0, |acc, c| {
+                *acc += c;
+                Some(*acc)
+            })
+            .collect_vec();
 
-        let mut children = vec![chunk_ends.into_array()];
+        let num_chunks = chunk_ends.len() - 1;
+        let length = (*chunk_ends.last().unwrap()) as usize;
+
+        let mut children = vec![PrimitiveArray::from_vec(chunk_ends, NonNullable).into_array()];
         children.extend(chunks);
 
-        Self::try_from_parts(dtype, ChunkedMetadata, children.into(), StatsSet::new())
+        Self::try_from_parts(
+            dtype,
+            length,
+            ChunkedMetadata { num_chunks },
+            children.into(),
+            StatsSet::new(),
+        )
     }
 
     #[inline]
     pub fn chunk(&self, idx: usize) -> Option<Array> {
+        let chunk_start = usize::try_from(&scalar_at(&self.chunk_ends(), idx).unwrap()).unwrap();
```

Comment:
There's a method on primitive array to get the scalar value out. Since we construct the ends array ourselves it should be safe to use it

Reply:
Not if it's been compressed

Reply:
hmm, ok, forgot this can be read